homeassistant/components/zwave_js/config_flow.py

```diff
@@ -784,7 +783,21 @@ async def async_step_on_supervisor(
                     {CONF_USE_ADDON: self.config_entry.data.get(CONF_USE_ADDON, True)}
                 ),
             )
+
         if not user_input[CONF_USE_ADDON]:
+            if self.config_entry.data.get(CONF_USE_ADDON):
+                # Unload the config entry before stopping the add-on.
+                await self.hass.config_entries.async_unload(self.config_entry.entry_id)
+                addon_manager = get_addon_manager(self.hass)
+                _LOGGER.debug("Stopping Z-Wave JS add-on")
+                try:
+                    await addon_manager.async_stop_addon()
```

Comment:
What about the "Start on boot" option? Can we disable that?

Reply:
That will need an extension to the add-on manager. I suggest we follow up with that in a separate PR. Currently the add-on manager only allows changing the add-on specific config of the add-on options.

We should also consider marking the add-on as system managed when the integration controls it.

Here's the whole add-on Supervisor API available:
https://developers.home-assistant.io/docs/api/supervisor/endpoints#add-ons

Reply:
@MindFreeze does my suggestion sound ok?